model.py

```diff
@@ -419,6 +419,13 @@ def library_stage(self, value):
     def pls_id(self):
         return ConfigurationSetting.for_library(Library.PLS_ID, self)
 
+    @property
+    def number_of_patrons(self):
+        # This is only meaningful if the library is in production.
+        if not self.in_production:
+            return '0'
+        return str(len(filter(lambda id: (id.type == 'Adobe Account ID'), self.delegated_patron_identifiers)))
```

Comment:
I'm concerned this is going to fall over when there are 100,000 delegated patron identifiers (as there are now for NYPL and Open eBooks). If it's easier to get the count for one library than for all libraries at once, then that should be fine, but the counting really ought to be done inside the database. Otherwise we're creating 100,000 Python objects just to count how many there are.

Reply:
Ah.  That is...a very good point.  Can you clarify what you mean about doing the counting inside the database, though?

Reply:
If you create a query and call count() on it, you'll execute a SELECT COUNT statement and the database will return the number of rows rather than the actual rows.

There are ways to do even fancier queries with the `select()` method -- `Work.to_search_document` is an extreme example and `CoverageRecord.bulk_add` is a less extreme example. If you're interested I'm happy to go over this with you so we can figure out how to do this for every library in one query. But building a query with SQLAlchemy and calling `count()` on it will get us where we need to be in terms of performance.